### PR TITLE
Refactor diffeq

### PR DESCRIPTION
**Pull request by pnkraemer**
steprule is now an argument for solve(), not an attribute of the solver. This is the first step of the refactoring, but independent enough that it can be merged independently.

### CODE REVIEW

File: src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ class GaussianIVPFilter(odesolver.ODESolver):
     further considerations to, e.g., BVPs.
     """
 
-    def __init__(self, ivp, gaussfilt, steprl):
+    def __init__(self, ivp, gaussfilt):
         """
         steprule : stepsize rule
```
Comment (nathanaelbosch) [inline src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py:24]: This line can be removed from the docstring

File: src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py [hunk 2]
```diff
@@ -37,9 +37,8 @@ def __init__(self, ivp, gaussfilt, steprl):
             raise ValueError("Please initialise a Gaussian filter with an ODEPrior")
         self.ivp = ivp
         self.gfilt = gaussfilt
-        odesolver.ODESolver.__init__(self, steprl)
 
-    def solve(self, firststep, **kwargs):
+    def solve(self, firststep, steprule, **kwargs):
         """
         Solve IVP and calibrates uncertainty according
         to Proposition 4 in Tronarp et al.
```
Comment (nathanaelbosch) [inline src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py:44]: `steprule` should be added to the docstring

**Review by nathanaelbosch — CHANGES_REQUESTED** [review]

**Review by nathanaelbosch — APPROVED** [review]

**Comment by JonathanWenger** [comment]
@pnkraemer Did you use admin privileges to merge this even though `black` is failing? Currently `black` is failing on master.

**Comment by pnkraemer** [comment]
not that I know of, but in any case black did not complain about this PR.

**Comment by nathanaelbosch** [comment]
This could again be due to a new Black release. I'll fix it by pinning the version.

**Pull request merged**